PP-1307: Test "test_sister_mom_crash" of TestSisterMom Fails as it is not able to find pbsdsh path while submitting the job

Mahalty opened the pull request:
<!--- Please review your changes in preview mode -->
<!--- Provide a general summary of your changes in the Title above -->

#### Bug/feature Description
* *[PP-1307](https://pbspro.atlassian.net/browse/PP-1307)*

#### Analysis
*  In the test we are submitting job as pbsuser with contents of the job script as "pbsdsh dd if=/dev/zero of=/dev/null". On the systems where the test failed, the test wasn't able to fetch the path for the command "pbsdsh", hence job didn't execute successfully.

#### Solution Description
* Test need to be updated with the path of pbsdsh command while submitting the job

#### Testing logs/output
* [Execution_logs_PP-1307_after_fix.txt](https://github.com/PBSPro/pbspro/files/2419499/Execution_logs_PP-1307_after_fix.txt)
* [Execution_logs_PP-1307_without_fix.txt](https://github.com/PBSPro/pbspro/files/2419500/Execution_logs_PP-1307_without_fix.txt)





#### Checklist:
<!--- Use the preview button to see the checkboxes/links properly. -->
<!--- Go over the following points, and put an `x` (without spaces around it) in the boxes that apply. -->
<!--- If you're unsure about any of these, don't hesitate to ask. We're here to help! -->
- [x] I have joined the **[pbspro community forum](http://community.pbspro.org/)**.
- [x] My pull request contains a **single, signed** commit. See **[setting up gpg signature](https://pbspro.atlassian.net/wiki/display/DG/Signing+Your+Git+Commits)**.
- [x] My code follows the **[coding style](https://pbspro.atlassian.net/wiki/display/DG/Coding+Standards)** of this project.
- [ ] My change requires project documentation. See **[required documentation checklist](https://pbspro.atlassian.net/wiki/display/DG/Checklist+for+Developing+Features+and+Bug+Fixes)** for details.
   - [ ] I have added documentation in the **[project documentation area](https://pbspro.atlassian.net/wiki/display/PD)**.
- [ ] I have added new **PTL test(s) to my commit**. (See **[using PTL for testing](https://pbspro.atlassian.net/wiki/display/DG/Using+PTL+for+Testing)**) *(or)*
   - [ ] I have added  **manual test(s) to this pull request and explained why PTL is not appropriate** for this case.
- [ ] All new and existing automated tests have passed. (See **[running automated PTL tests](https://pbspro.atlassian.net/wiki/display/DG/PTL+Quick+Start+Guide)**).
- [x] I have attached **test logs to this pull request** as evidence of testing/verification.


__***For further information please visit the [Developer Guide Home](https://pbspro.atlassian.net/wiki/display/DG/Developer+Guide+Home).***__

Mahalty commented:
Thanks @VishrutArora 
@PBSPro/pbspro-maintainers Could any one of you please review this PR

bhroam commented on test/tests/functional/pbs_sister_mom_crash.py:
```diff
@@ -53,7 +53,8 @@ def test_sister_mom_crash(self):
         sister_mom = self.moms.keys()[1]
-        script = "pbsdsh dd if=/dev/zero of=/dev/null"
+        pbsdsh_path = self.du.which(exe='pbsdsh')
+        script = "%s dd if=/dev/zero of=/dev/null" % pbsdsh_path
```
The problem with which() is that it mainly looks through the path to find a program.  We know where pbsdsh is.  Just join pbs_conf['PBS_EXEC'] with bin and pbsdsh

Mahalty commented:
Thanks @bhroam for the review. I have addressed your comments. Please have a look
[Latest_Execution_logs_PP-1307.txt](https://github.com/PBSPro/pbspro/files/2432976/Latest_Execution_logs_PP-1307.txt)

bhroam commented on test/tests/functional/pbs_sister_mom_crash.py:
```diff
@@ -53,7 +53,9 @@ def test_sister_mom_crash(self):
-        pbsdsh_path = self.du.which(exe='pbsdsh')
+        conf = self.du.parse_pbs_config()
+        pbs_exec = conf['PBS_EXEC']
+        pbsdsh_path = os.path.join(pbs_exec, "bin", "pbsdsh")
```
I don't think you need to parse pbs.conf.  I think self.server.pbs_conf['PBS_EXEC']

Mahalty commented:
Latest execution logs:
[Latest_Execution_logs_1_PP-1307.txt](https://github.com/PBSPro/pbspro/files/2440944/Latest_Execution_logs_1_PP-1307.txt)

Mahalty commented:
Thanks @bhroam for the review.

@PBSPro/pbspro-maintainers: I have squashed my commits and rebased my branch. Could any one of you please take this PR forward
